### PR TITLE
feat: telemetry config

### PR DESCRIPTION
**Pull request by mjuniper**
1. Description: Migrate site telemetry config; add json-schema and ui-schema for site telemetry.

1. Related: [8104](https://devtopia.esri.com/dc/hub/issues/8104)

1. [X] Updated meaningful TSDoc to methods including Parameters and Returns, see [Documentation Guide](https://esri.github.io/hub-components/storybook/?path=/story/guides-documentation--page)

1. [X] used semantic commit messages
  
1. [X] PR title follows semantic commit format (**CRITICAL** if the title is not in a semantic format, the release automation will not run!)

1. [X] updated `peerDependencies` as needed. **CRITICAL** our automated release system can **not** be counted on to update `peerDependencies` so we _must_ do it manually in our PRs when needed. See the [updating peerDependencies](/RELEASE.md#Updating-peerDependencies) section of the release instructions for more details.

### CODE REVIEW

File: packages/common/src/core/schemas/internal/getEntityEditorSchemas.ts [hunk 1]
```diff
@@ -63,6 +63,8 @@ export async function getEntityEditorSchemas(
           import("../../../sites/_internal/SiteUiSchemaFollowers"),
         "hub:site:discussions": () =>
           import("../../../sites/_internal/SiteUiSchemaDiscussions"),
+        "hub:site:settings": () =>
+          import("../../../sites/_internal/SiteUiSchemaTelemetry"),
```
Comment (juliannemarik) [inline packages/common/src/core/schemas/internal/getEntityEditorSchemas.ts:67]: wondering if this `uiSchema` file should be called `SiteUiSchemaSettings`?

File: packages/common/src/sites/_internal/_migrate-telemetry-config.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  if (getProp(model, "item.properties.schemaVersion") >= newSchemaVersion) {
+    return model;
+  }
+
+  // apply migration
+  const clone = cloneObject(model);
+
+  clone.data.telemetry = {};
+  // get allowPrivacyConfig from consentNotice capability
+  const capabilities = getProp(model, "data.values.capabilities") || [];
```
Comment (juliannemarik) [inline packages/common/src/sites/_internal/_migrate-telemetry-config.ts:25]: wondering if we should convert this capability to a permissions system feature like we recently did for the "hideFollow" and "disableDiscussions" capabilities? 

Essentially you'd need to: 
1. add a new feature flag to the `SiteDefaultFeatures` [here](https://github.com/Esri/hub.js/blob/feat/7735-advanced-json-editor/packages/common/src/sites/_internal/SiteBusinessRules.ts#L10-L11)
2. add a mapping for the capability to feature [here](https://github.com/Esri/hub.js/blob/feat/7735-advanced-json-editor/packages/common/src/sites/_internal/capabilities/capabilityToFeatureMap.ts)

The `migrateLegacyCapabilitiesToFeatures` and `convertFeaturesToLegacyCapabilites` functions should handle the rest

Then here, you'd reference `getProp(model, "data.settings.features")` instead

Comment (mjuniper) [reply packages/common/src/sites/_internal/_migrate-telemetry-config.ts:25]: Well, this is something we hope to be used across the platform so i think we really want it to be at data.telemetry.consentNotice.allowPrivacyConfig

**Review by juliannemarik — APPROVED** [review]

File: packages/common/src/sites/_internal/_migrate-telemetry-config.ts (hunk 2 above)
Comment (dbouwman) [reply packages/common/src/sites/_internal/_migrate-telemetry-config.ts:25]: Additionally, this config is passed into telemetry.js, and everything else is handled internally - the app does not need to know anything about the privacy settings in terms of permissions

**Review by dbouwman — APPROVED** [review]

**Pull request merged**